lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -3357,7 +3362,14 @@ CheckedCastKind TypeChecker::typeCheckCheckedCast(Type fromType,
         const auto &fromElt = fromTuple->getElement(i);
         const auto &toElt = toTuple->getElement(i);
 
-        if (fromElt.getName() != toElt.getName())
+        // We only should perform name validation if both element have name
+        // bacause unlabeled tuple elements can be converted to labeled ones
```

Comment:
```suggestion
        // We should only perform name validation if both element have a label,
        // because unlabeled tuple elements can be converted to labeled ones
```

Reply:
Fixed! Thanks :)